src/common/framework/cfgTools.ml

```diff
@@ -405,24 +405,24 @@ let createCFG (file: file) =
                   (* scc has no successors but also doesn't contain return node, requires additional connections *)
                   (* find connection candidates from loops *)
                   let targets =
-                    NH.keys scc.nodes
-                    |> BatEnum.concat_map (fun fromNode ->
+                    NH.to_seq_keys scc.nodes
+                    |> Seq.concat_map (fun fromNode ->
                         NH.find_all loop_head_neg1 fromNode
-                        |> BatList.enum
-                        |> BatEnum.filter (fun toNode ->
+                        |> List.to_seq
+                        |> Seq.filter (fun toNode ->
                             not (NH.mem scc.nodes toNode) (* exclude candidates into the same scc, those wouldn't help *)
                           )
-                        |> BatEnum.map (fun toNode ->
+                        |> Seq.map (fun toNode ->
                             (fromNode, toNode)
                           )
                       )
-                    |> BatList.of_enum
+                    |> List.of_seq
                   in
                   let targets = match targets with
                     | [] ->
                       let scc_node =
-                        NH.keys scc.nodes
-                        |> BatList.of_enum
+                        NH.to_seq_keys scc.nodes
+                        |> BatList.of_seq
                         |> BatList.min ~cmp:Node.compare (* use min for consistency for incremental CFG comparison *)
```

Comment:
It's a bit silly to turn it into a list just to find the minimum, but there isn't `Seq.min` (or `BatSeq.min` with `~cmp`).
It's not worse than before, but leaving a TODO here would be good.